### PR TITLE
Collector send to kafka

### PR DESCRIPTION
**Pull request by lynxbyorion**

### CODE REVIEW

File: src/store.rs [hunk 1]
```diff
@@ -37,56 +40,168 @@ impl Chunk {
...
-                    is_send = true
+    // return requests if chunk is full
+    pub fn push(&mut self, request: Request) -> Option<(String, Vec<Request>)> {
+        match self.hosts_to_topics.get(&request.host) {
+            Some(topic) => match self.chunks_by_topics.get_mut(topic) {
+                Some(chunk) => {
+                    if chunk.is_full(&request) {
+                        return Option::Some((topic.clone(), chunk.pop_all()));
+                    }
+                    chunk.push(request.clone());
```
Comment (stuffy-idler) [inline src/store.rs:82]: Request will be skipped if the chunk is full. Please, add test for this case

File: src/store.rs [hunk 2]
```diff
@@ -37,56 +40,168 @@ impl Chunk {
...
+            ..init_service_config()
+        });
+
+        assert!(store.push(Request { ..init_request() }).is_none())
+    }
+
+    #[test]
+    fn return_chunk_by_max_len_chunk() {
+        let mut store = Store::new(&ServiceConfig {
+            max_len_chunk: 2,
```
Comment (stuffy-idler) [inline src/store.rs:129]: I guess it will be overridden by `..init_service_config()`

**Review by stuffy-idler — CHANGES_REQUESTED** [review]

File: src/main.rs [hunk 3]
```diff
@@ -28,88 +32,125 @@ async fn main() -> std::io::Result<()> {
...
+    if let Err(e) = state.sender.send(r).await {
         error!("sender: {}", e)
     };
 
     "OK"
 }
 
-async fn process(mut reciver: mpsc::Receiver<store::Request>, config: &AppConfig) {
-    let mut store = store::Store::new(config.service.max_size_chunk, config.service.max_len_chunk);
+async fn service_process(
```
[the remark this answers is not in the document]
Comment (lynxbyorion) [reply src/main.rs]: Done.

File: src/main.rs [hunk 4]
```diff
@@ -28,88 +32,125 @@ async fn main() -> std::io::Result<()> {
...
+    mut reciver: mpsc::Receiver<store::Request>,
```
[the remark this answers is not in the document]
Comment (lynxbyorion) [reply src/main.rs]: Fixed.

File: src/main.rs [hunk 5]
```diff
@@ -28,88 +32,125 @@ async fn main() -> std::io::Result<()> {
...
+                    }
                     delay.reset();
                 };
-                println!("add")
             }
 
             _ = delay.tick() => {
-                store.send();
-                println!("click")
+                for (topic, requests) in store.pop_all() {
```
[the remark this answers is not in the document]
Comment (lynxbyorion) [reply src/main.rs]: Rewrite.

File: src/main.rs [hunk 6]
```diff
@@ -28,88 +32,125 @@ async fn main() -> std::io::Result<()> {
...
+    mut reciver: mpsc::Receiver<(String, Vec<Request>)>,
+    config: AppConfig,
+) -> Result<(), KafkaError> {
+    let mut producer = Producer::from_hosts(config.service.kafka_brokers)
+        .with_ack_timeout(Duration::from_secs(1))
+        .with_required_acks(RequiredAcks::One)
+        .create()?;
+
+    while let Some((topic, requests)) = reciver.recv().await {
+        let s = serde_json::to_string(&requests).unwrap();
```
[the remark this answers is not in the document]
Comment (lynxbyorion) [reply src/main.rs]: Added handler.

**Review by stuffy-idler — APPROVED** [review]

**Pull request merged**